includes/embeds/class-amp-gutenberg-categories-handler.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Class AMP_Gutenberg_Categories_Handler
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Gutenberg_Categories_Handler
+ *
+ * @since 1.0
+ */
+class AMP_Gutenberg_Categories_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Register embed.
+	 */
+	public function register_embed() {
+		if ( function_exists( 'gutenberg_init' ) ) {
+			add_action( 'the_post', array( $this, 'override_category_block_render_callback' ) );
+		}
+	}
+
+	/**
+	 * Unregister embed.
+	 */
+	public function unregister_embed() {}
```

Comment:
This should undo what is done in `register_embed`. The original `render_callback` for the `core/categories` block should be saved and then restored here.